src/components/plots/stackedAreaPlot.ts

```diff
@@ -137,26 +135,27 @@ export module Plots {
         return;
       }
 
-      var orientation = this._isVertical ? "vertical" : "horizontal";
-      var keyAccessor = StackedPlotUtils.keyAccessor(this, orientation);
-      var valueAccessor = StackedPlotUtils.valueAccessor(this, orientation);
-      var datasetKeys = this._datasetKeysInOrder;
-      var keyToPlotDatasetKey = this._key2PlotDatasetKey;
-      var filter = this._filterForProperty(this._isVertical ? "y" : "x");
+      var datasets = this.datasets();
+      var keyAccessor = this.x().accessor;
+      var valueAccessor = this.y().accessor;
+      var filter = this._filterForProperty("y");
 
-      StackedPlotUtils.checkSameDomainForStacks(keyAccessor, datasetKeys, keyToPlotDatasetKey);
+      this._checkSameDomain(datasets, keyAccessor);
+      this._stackOffsets = Utils.Stacked.computeStackOffsets(datasets, keyAccessor, valueAccessor);
+      this._stackedExtent = Utils.Stacked.computeStackExtent(datasets, keyAccessor, valueAccessor, this._stackOffsets, filter);
+    }
 
-      var stackOffsets = StackedPlotUtils.computeStackOffsets.call(this, keyAccessor, valueAccessor, datasetKeys, keyToPlotDatasetKey);
+    private _checkSameDomain(datasets: Dataset[], keyAccessor: Accessor<any>) {
+      var keySets = datasets.map((dataset) => {
+        return d3.set(dataset.data().map((datum, i) => keyAccessor(datum, i, dataset).toString())).values();
+      });
+      var domainKeys = Utils.Stacked.getDomainKeys(datasets, keyAccessor);
```

Comment:
I'm unsure if this warrants making an endpoint for.  We can just calculate it here